packages/@react-stately/layout/src/ListViewLayout.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import {LayoutInfo, Rect, Size} from '@react-stately/virtualizer';
+import {LayoutNode, ListLayout} from './ListLayout';
+
+export class ListViewLayout<T> extends ListLayout<T> {
+  buildCollection(): LayoutNode[] {
+    let y = this.padding;
+    let nodes = [];
+    for (let node of this.collection) {
+      let layoutNode = this.buildChild(node, 0, y);
+      y = layoutNode.layoutInfo.rect.maxY;
+      nodes.push(layoutNode);
+    }
+
+    // TODO: if supporting infinite scroll, will need to not take up the whole list
+    if (this.isLoading) {
```

Comment:
Doesn't ListLayout already support the loader?

Reply:
ListLayout caps the loading height at 40: 
https://github.com/adobe/react-spectrum/blob/main/packages/@react-stately/layout/src/ListLayout.ts#L148

The ListViewLayout has it so that it takes the whole `visibleRect.height` so that it's veritically centered.

Reply:
If we want to avoid creating a layout, should we have some option for it to span the whole visibleRect?